src/main/java/com/pimp/controller/CalendarController.java

```diff
@@ -67,4 +71,44 @@ public void addEvent(@Valid @RequestBody Event event, @PathVariable String calen
     }
   }
 
+  @PreAuthorize("#oauth2.hasScope('user_actions')")
+  @RequestMapping(method = PUT, path = "/event/{eventKey}")
+  public void editEvent(@Valid @RequestBody Event event, Principal principal) {
+    Optional<Calendar> optionalCalendar =
+        calendarService.getEventContainingCalendarByUserName(principal.getName(), event.getKey());
+
+    if (!optionalCalendar.isPresent()) {
+      throw new EntityNotFoundException("An event with the key " + event.getKey() +
+          " does not exist for the user");
+    } else {
+      List<Event> events = optionalCalendar.get().getEvents()
+        .stream()
+        .map(aEvent -> aEvent.getKey().equals(event.getKey()) ? event : aEvent)
+        .collect(Collectors.toList());
+      Calendar calendar = optionalCalendar.get();
+      calendar.setEvents(events);
+      calendarService.save(calendar);
+    }
+  }
+
+  @PreAuthorize("#oauth2.hasScope('user_actions')")
+  @RequestMapping(method = DELETE, path = "/event/{eventKey}")
+  public void deleteEvent(@Valid @RequestBody Event event, Principal principal) {
+    Optional<Calendar> optionalCalendar =
+        calendarService.getEventContainingCalendarByUserName(principal.getName(), event.getKey());
```

Comment:
Especially because `getEventContainingCalendarByUserName()` calls `getCalendarsByUser()` which queries ALL calendars from the database and let's the JVM filter the results instead of letting MongoDB (which probably is much better at it) do the gruntwork of filtering.